quarkchain/cluster/subscription.py

```diff
@@ -42,17 +42,30 @@ def remove_subscriber(self, sub_id):
         raise InvalidParams("subscription not found")
 
     async def notify_new_heads(self, header: MinorBlockHeader):
-        # TODO
-        data = None
+        from quarkchain.cluster.jsonrpc import minor_block_header_encoder
+
+        data = minor_block_header_encoder(header)
         await self.__notify(SUB_NEW_HEADS, data)
 
     async def notify_new_pending_tx(self, tx_hash: bytes):
         await self.__notify(SUB_NEW_PENDING_TX, "0x" + tx_hash.hex())
 
-    async def notify_log(self, height: int):
-        # TODO
-        data = None
-        await self.__notify(SUB_LOGS, data)
+    async def notify_log(
+        self, end_block_header: MinorBlockHeader, is_removed: Optional[bool] = False
```

Comment:
should also expect a size parameter, then we can batch multiple log notification to 1